activerecord/lib/active_record/core.rb

```diff
@@ -199,11 +199,8 @@ def find(*ids) # :nodoc:
                                  name, primary_key)
       end
 
-      def find_by(*args) # :nodoc:
-        return super if scope_attributes? || !(Hash === args.first) || reflect_on_all_aggregations.any?
-
-        hash = args.first
-
+      def find_by(hash, *args) # :nodoc:
+        return super if scope_attributes? || !(Hash === hash) || reflect_on_all_aggregations.any?
         return super if hash.values.any? { |v|
           v.nil? || Array === v || Hash === v || Relation === v
```

Comment:
Noted in passing: this sounds like it would let a Range through, but the below seems ill-equiped to handle that.